api/commands/compile.ts

```diff
@@ -89,26 +89,17 @@ export class CompileChildProcess {
 
   public async compile(compileConfig: dataform.ICompileConfig) {
     const compileInChildProcess = new Promise<string>(async (resolve, reject) => {
-      // Handle any Error caused by spawning the child process, or sent directly from the child process.
       this.childProcess.on("error", (e: Error) => reject(coerceAsError(e)));
-      this.childProcess.on("message", (e: Error) => reject(coerceAsError(e)));
-
-      // Handle UTF-8 string chunks returned by the child process.
-      const pipe = this.childProcess.stdio[4] as Readable;
-      const chunks: Buffer[] = [];
-      pipe?.on("readable", () => {
-        let buffer: Buffer = pipe.read();
-        while (buffer) {
-          chunks.push(buffer);
-          buffer = pipe.read();
+
+      this.childProcess.on("message", (messageOrError: string | Error) => {
+        if (typeof messageOrError === "string") {
+          resolve(messageOrError);
```

Comment:
bring back the `return` after this, it's not really valid to both `resolve` and `reject` a single `Promise`

Reply:
Ah I thought resolve and reject returned